tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -418,5 +440,39 @@ private static async Task GetIntelligentTestRunnerSkippableTestsAsync()
                 Log.Error(ex, "ITR: Error getting skippeable tests.");
             }
         }
+
+        private static bool IsEventPlatformProxySupportedByAgent(ref IDiscoveryService discoveryService)
+        {
+            var eventPlatformProxyEnabled = false;
+            discoveryService = DiscoveryService.Create(new ImmutableExporterSettings(_settings.TracerSettings.Exporter));
+            AgentConfiguration? agentConfiguration = null;
+            ManualResetEventSlim manualResetEventSlim = new(false);
+            LifetimeManager.Instance.AddShutdownTask(() => manualResetEventSlim.Set());
+
+            Log.Debug("Waiting for agent configuration...");
+            var discoveryServiceCopy = discoveryService;
+            discoveryServiceCopy.SubscribeToChanges(CallBack);
+            void CallBack(AgentConfiguration aConfiguration)
+            {
+                agentConfiguration = aConfiguration;
+                manualResetEventSlim.Set();
+                discoveryServiceCopy.RemoveSubscription(CallBack);
+            }
```

Comment:
What's with the `discoveryServiceCopy` 🤔  Why can't we create an instance and pass it into this method?

Reply:
`discoveryService` is a `ref` parameter and cannot be used as a closure (in `CallBack()`) that's why I create a copy.

Reply:
Makes sense, but again, why not create the discovery service in the caller instead and pass in a normal ref?

Reply:
true!, done in https://github.com/DataDog/dd-trace-dotnet/pull/3322/commits/d5c4a023622d2f1eed62a10f01956bf3a51ab1cc